### PR TITLE
docs: Add and format descriptions of the public API's class, methods, and parameters. (EDU-2453)

### PR DESCRIPTION
**Pull request by jamesneimanconsulting**
docs: Add and format descriptions of the public API's class, methods, and parameters. (EDU-2453)
Provide missing descriptions and make the wording consistent with readme.io.

### CODE REVIEW

**Comment by jamesneimanconsulting** [comment]
@thomaszurkan-optimizely , @jonslaught Please review and merge if approved.

File: core-api/src/main/java/com/optimizely/Optimizely.java [hunk 1]
```diff
@@ -0,0 +1,1184 @@
+/****************************************************************************
```
Comment (mikeng13) [inline core-api/src/main/java/com/optimizely/Optimizely.java]: This file shouldn't exist. The Optimizely class is in the path: core-api/src/main/java/com/optimizely/ab/Optimizely.java

**Comment by jamesneimanconsulting** [comment]
@mikeng13 I've added a commit to this PR.

**Comment by jamesneimanconsulting** [comment]
@jonslaught I've updated this PR with a new commit for consistency and a cross-project effort to align wording, grammar,  etc.

**Comment by jamesneimanconsulting** [comment]
@mikeng13 , @jonslaught Merge conflicts resolved

**Comment by mikeng13** [comment]
**Obsolte**

**Pull request closed without merging**